core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -76,6 +76,14 @@ package object config {
     .timeConf(TimeUnit.MILLISECONDS)
     .createWithDefaultString("3s")
 
+  private[spark] val DYN_ALLOCATION_RECOVER_CACHE =
+    ConfigBuilder("spark.dynamicAllocation.recoverCachedData").booleanConf.createWithDefault(false)
```

Comment:
`.cacheRecovery.enabled` to follow pattern of other feature flags.

I also kinda prefer multiple lines in these constant declarations, for readability.

Reply:
fixed